robottelo/ui/container.py

```diff
@@ -0,0 +1,160 @@
+from robottelo.ui.navigator import Navigator
+
+
+class Container(Base):
+    """Provides the CRUD functionality for Docker Containers."""
+
+    def _configure_orgs(self, orgs, org_select):
+        """Provides configuration capabilities for docker container
+        organization. The following format should be used::
+
+            orgs=['Aoes6V', 'JIFNPC'], org_select=True
+
+        """
+        self.configure_entity(
+            orgs,
+            FILTER['container_org'],
+            tab_locator=tab_locators['tab_org'],
+            entity_select=org_select,
+        )
+
+    def _configure_locations(self, locations, loc_select):
+        """Provides configuration capabilities for docker container location
+
+        The following format should be used::
+
+            locations=['Default Location'], loc_select=True
+
+        """
+        self.configure_entity(
+            locations,
+            FILTER['container_loc'],
+            tab_locator=tab_locators['tab_loc'],
+            entity_select=loc_select,
+        )
+
+    def _form_locator_name(self, partial_locator):
+        """Form locator using provided friendly UI name, e.g. 'Content View'"""
+        return '.'.join((
+            'container',
+            (partial_locator.lower()).replace(' ', '_')
+        ))
+
+    def create(self, resource_name, name, command, parameter_list):
+
+        """Creates a docker container. All values should be passed in absolute
+        correspondence to UI. Parameters names created in self-descriptive
+        manner. Of course, we can easily expand list of parameters and create
+        custom flows for specific situations. Here are some examples of
+        parameter_list values from each main tab::
+
+            [
+                {'main_tab_name': 'Preliminary', 'sub_tab_name': 'Location',
+                'name': ['Default Location']},
+                {'main_tab_name': 'Image', 'sub_tab_name': 'Content View',
+                'name': 'Lifecycle Environment', 'value': self.lce.name},
+                {'main_tab_name': 'Image', 'sub_tab_name': 'Docker Hub',
+                'name': 'Docker Hub Tag', 'value': 'latest'},
+                {'main_tab_name': 'Configuration', 'name': 'Memory',
+                'value': '512m'},
+                {'main_tab_name': 'Environment', 'name': 'TTY', 'value': True},
+            ]
+        """
+        self.assign_value(locators['container.resource_name'], resource_name)
+        for parameter in parameter_list:
+            if parameter['main_tab_name'] == 'Preliminary':
+                if parameter['sub_tab_name'] == 'Organization':
+                    self._configure_orgs(parameter['name'], True)
+                elif parameter['sub_tab_name'] == 'Location':
+                    self._configure_locations(parameter['name'], True)
+        self.click(locators['container.next_section'])
+        current_tab = self._form_locator_name('Content View Tab')
+        for parameter in parameter_list:
```

Comment:
this function is looking too complex for readability.  Is there a  necessity to call the following loop 4 times?  Can it be achieved in one loop?
```py
for parameter in parameter_list:
```